src/main/kotlin/com/hibob/academy/dao/OwnerDao.kt

```diff
@@ -0,0 +1,50 @@
+package com.hibob.academy.dao
+
+import org.jooq.DSLContext
+import org.jooq.Record
+import org.jooq.RecordMapper
+import org.springframework.stereotype.Component
+import java.math.BigInteger
+import java.time.LocalDateTime
+import java.util.Date
+
+data class Owner(val name: String, val companyId: Long, val employId: String)
+
+
+@Component
+class OwnerDao(private val sql: DSLContext) {
+    private val table = OwnerTable.instance
+
+
+    private val ownerMapper = RecordMapper<Record, Owner> { record ->
+        Owner(record[table.name], record[table.companyId], record[table.employId])
+    }
+
+    fun getOwners(): List<Owner> =
+        sql.select(table.name, table.companyId, table.employId)
+            .from(table)
+            .fetch(ownerMapper)
+
+
+    fun createOwner(owner : Owner) : Long {
+        val res = sql.insertInto(table)
+            .set(table.name, owner.name)
+            .set(table.companyId, owner.companyId)
+            .set(table.employId, owner.employId)
+            .onConflict(table.companyId, table.employId)
+            .doNothing()
+            .returningResult(table.id)
+            .fetchOne()
+
+        return res?.get(table.id) ?: 0
```

Comment:
you can use !! here to get the ID since we are sure that we have an id here

Reply:
i did here without and return 0 because when i create an owner i return the id that the owner get from the table (because the serial) so when i try to create another owner with the same company id and employ id i got 0 in this case . i dont know if its the right way - to get the id when i create an owner ? - in the pet the same reason

Reply:
You don't have to use onConflict, that will cause an exception.
That way you will get the right id if a owner was created